pkg/app/piped/executor/waitapproval/waitapproval.go

```diff
@@ -161,3 +185,37 @@ func (e *Executor) getMentionedAccounts(event model.NotificationEventType) ([]st
 
 	return notification.FindSlackAccounts(event), nil
 }
+
+func (e *Executor) validateApproverNum(ctx context.Context, approver string, minApproverNum int) (string, bool) {
+	if minApproverNum <= 1 {
+		e.LogPersister.Infof("Got approval from %q", approver)
+		return approver, true
+	}
+	as, ok := e.MetadataStore.Stage(e.Stage.Id).Get(approversKey)
+	if !ok {
+		e.LogPersister.Infof("Got approval from %q", approver)
+		e.LogPersister.Infof("Waiting for %d other approvers...", minApproverNum-1)
+		if err := e.MetadataStore.Stage(e.Stage.Id).Put(ctx, approversKey, approver); err != nil {
+			e.LogPersister.Errorf("Unabled to save approver information to deployment, %v", err)
+		}
+		return approver, false
+	}
+
+	for _, a := range strings.Split(as, ", ") {
+		if a == approver {
+			e.LogPersister.Infof("Approval from the same user (%s) will not be counted", approver)
+			return "", false
+		}
+	}
+	e.LogPersister.Infof("Got approval from %q", approver)
+
+	totalAs := fmt.Sprintf("%s, %s", as, approver)
+	if c := minApproverNum - len(strings.Split(totalAs, ", ")); c > 0 {
```

Comment:
Again, doesn't it go well with this?
```suggestion
	if minApproverNum > len(strings.Split(totalAs, ", ")) {
```

Reply:
https://github.com/pipe-cd/pipe/pull/2802/files#diff-02a6821aa5f1210155af9610f8bb1b7683ed309d3ad56a533f0505d12ae6ffd6R214

Sorry, I forgot replying to this one. `c` is used inside of if block.

Reply:
makes sense